modules/apigee_edge_apidocs/apigee_edge_apidocs.info.yml

```diff
@@ -8,3 +8,4 @@ dependencies:
 - drupal:text
 - drupal:file
 - apigee_edge:apigee_edge
+- apigee_edge:deprecated
```

Comment:
This makes the module uninstallable in the install screen before the requirements is ever reached.

<img width="973" alt="image" src="https://user-images.githubusercontent.com/159693/58138330-7c54b280-7bea-11e9-9e5f-a37f43258960.png">

Reply:
Well, this is rather seems to be some kind of hack then a proper solution ;S Have you seen this doing by other modules?
What I originally suggested on the sprint planning call is an "install" phase check in `apigee_edge_apidocs_requirements()` that does not allow to enable the module anymore and displays a deprecation message. Isn't that possible?

Reply:
Actually what you have added to `apigee_edge_apidocs_requirements()` is perfectly enough.

![Peek 2019-05-22 08-10](https://user-images.githubusercontent.com/1755573/58151116-05123380-7c69-11e9-88bd-0c83b7145f66.gif)

Reply:
Meh... so there is an issue with Drush... It does not run `hook_requirements()`.
https://github.com/drush-ops/drush/issues/3669

Reply:
The closest solution that I got is this:

```
<?php
/**
 * Implements hook_module_preinstall().
 */
function apigee_edge_apidocs_module_preinstall($module) {
  if (PHP_SAPI === 'cli' && $module === 'apigee_edge_apidocs') {
    // Workaround for a bug in Drush: https://github.com/drush-ops/drush/issues/3669
    \Drupal::service('module_installer')->uninstall(['apigee_edge_apidocs']);
    throw new RuntimeException('The Apigee Edge Apidocs module is deprecated and will be removed before Apigee Edge version 8.x-1.0 is released. Please use the Apigee API Catalog module as an alternative: https://www.drupal.org/project/apigee_api_catalog.');
  }
}
```

```sh
cli-drupal:/app$ drush pmu apigee_edge_apidocs -y
 [success] Successfully uninstalled: apigee_edge_apidocs
cli-drupal:/app$ drush en apigee_edge_apidocs -y

In apigee_edge_apidocs.module line 53:
                                                                                                                                                                                                                       
  The Apigee Edge Apidocs module is deprecated and will be removed before Apigee Edge version 8.x-1.0 is released. Please use the Apigee API Catalog module as an alternative: https://www.drupal.org/project/apigee_  
  api_catalog.                                                                                                                                                                                                         
                                                                                                                                                                                                                       

pm:enable [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-d|--debug] [-y|--yes] [--no] [--remote-host REMOTE-HOST] [--remote-user REMOTE-USER] [-r|--root ROOT] [-l|--uri URI] [--simulate] [--pipe] [-D|--define DEFINE] [--druplicon] [--xh-link XH-LINK] [--notify [NOTIFY]] [--] <command> [<modules>]...

cli-drupal:/app$ drush pm:list | grep apigee_edge_apidocs
  Apigee (Deprecated)                  Apigee Edge API Docs (apigee_edge_apidocs)                              Disabled   8.x-1.0-rc4   
```

At least Drupal Console calls `hook_requirements()`.

Reply:
Does drush 9 still try to install the module if you keep `- apigee_edge:deprecated`?